StockServiceFactory/src/main/java/edu/mbrooks/advancedjava/StockFactory/StockFactory.java

```diff
@@ -49,24 +57,178 @@ public void addQuote(StockQuote q) {
         this.quote = q; // the latest quote is always available...
     }
 
+    /**
+     *  This is a class private method that returns if the date on the quote is within the specifiec date range.
+     *
+     * @param chkDate   - is the date to compare
+     * @param from      - start of the date range
+     * @param until     - end of the date range
+     * @return boolean  - either is or isn't in the date range
+     */
+    private boolean isQuoteInDateRange(Calendar chkDate, Calendar from, Calendar until) {
+
+        boolean returnValue = false;
+
+        if (chkDate.after(from) && chkDate.before(until)) {
+            returnValue = true;
+        } else if (chkDate.compareTo(from) == 0) {
+            returnValue = true;
+        } else if (chkDate.compareTo(until) == 0) {
+            returnValue = true;
+        }
+        return returnValue;
+    }
+
+    /**
+     *
+     * This is special adaptation of the Annual interval
```

Comment:
I think you might have over engineered the interval concept. 
Basically, interval is  used to specify how many quotes you want for a given date range. 
say the date range is 1 week 
then the interval is one hour 
then you'd get back 24 * 7 quotes. 
does that make sense?

Reply:
See the previous comment - If I were tracking a stock, and I want to see historic performance, I may want 1 quote per year if the date range it the last 5 years. I can also look at quarterly performance, as well as annual performance.

Perhaps it is  over engineered, but it made perfect sense to me (i.e. Apple has been around for a long time and if I want to see how it has grown over the years, I need lots of info - the quote price, splits, etc. for every year)

I think over engineering would've been trying to determine if I needed the first or last quote of the quarter (I can actual see where that may make a difference depending on the perspective the user wants to take).